tools/lsp/common.rs

```diff
@@ -28,6 +28,26 @@ pub fn extract_element(node: SyntaxNode) -> Option<syntax_nodes::Element> {
     }
 }
 
+pub fn find_element_with_decoration(element: &syntax_nodes::Element) -> SyntaxNode {
```

Comment:
I'd rather keep these kind of helper functions that are only used for the preview under the preview module.